src/js/ValidationProgress.jsx

```diff
@@ -49,7 +49,7 @@ class ValidationProgress extends React.Component {
   }
 }
 
-ValidationProgress.propTypes = {initialCode: React.PropTypes.number};
-ValidationProgress.defaultProps = {initialCode: 3, appStatus: {get: function(){}, set: function(){}}};
+ValidationProgress.propTypes = {initialCode: React.PropTypes.number, appStatus: React.PropTypes.objectOf(React.PropTypes.func).isRequired};
```

Comment:
Minor but this would be easier to read if broken down:
```javascript
{
  initialCode: React.PropTypes.number,
  appStatus: React.PropTypes.objectOf(React.PropTypes.func).isRequired
}
```